doc/scheduled-configurations/scheduled-configurations-hld.md

```diff
@@ -0,0 +1,547 @@
+4. `timerangemgrd` will then create the crontab files used by the `cron` application to activate, or deactivate the status of the `TIME_RANGE_STATUS_TABLE`
+
+![scheduled configuration flow -- set](images/scheduled-configuration-flow-set.png)
+
+1. Apply scheduled configurations to the `SCHEDULED_CONFIGURATIONS` table in `CONFIG_DB`
+2. Changes to `SCHEDULED_CONFIGURATIONS` will cause a subscription notification to `scheduledconfigmgrd`
+3. If the time range this configuration is already active, it will publish this new configuration to the `APPL_DB`
+4. `scheduledconfigmgrd` saves the configuration internally
+
+#### Delete
+
+![time range flow -- delete](images/time-range-flow-del.png)
+
+1. A time range is deleted from the `TIME_RANGE` table in `CONFIG_DB`
+2. A subscription notification is sent to `timerangemgrd` with the op *delete*
+3. `timerangemgrd` deletes the crontab files associated with this time range
+4. `timerangemgrd` deletes the time range entry from the `TIME_RANGE_STATUS_TABLE` table found in the `STATE_DB`
+5. A subscription notification is sent to `scheduledconfigmgrd` the op *delete*
+6. `scheduledconfigmgrd` checks if the time range is active, if yes then it will deactivate the configurations bound to this time range by deleting the configurations from the `APPL_DB`
+7. Finally, `scheduledconfigmgrd` deletes the time range internally, and the scheduled configurations will be unbound
+
+![scheduled configuration flow -- delete](images/scheduled-configuration-flow-del.png)
+
+1. A scheduled configuration is deleted form the `SCHEDULED_CONFIGURATIONS` table in `CONFIG_DB`
+2. A subscription notification is sent to `scheduledconfigmgrd` with the op *delete*
+3. `scheduledconfigmgrd` checks if the scheduled configuration is active, if yes then it will deactivate the configuration by deleting the configurations from the `APPL_DB`
+4. Finally, `scheduledconfigmgrd` deletes the scheduled configuration internally
+
+#### Events
+
+![scheduled configuration flow -- events](images/scheduled-configuration-flow-events.png)
+
+1. During the scheduled activation time, `cron` updates the **status** field in the `TIME_RANGE_STATUS_TABLE` found in the `STATE_DB` to *active*
+2. A subscription notification is sent to `scheduledconfigmgrd`
+3. `scheduledconfigmgrd` applies all the configurations bound to the time range by publishing to the `APPL_DB`
+4. During the scheduled deactivation time, `cron` updates the **status** field in the `TIME_RANGE_STATUS_TABLE` found in the `STATE_DB` to *inactive*
+5. A subscription notification is sent to `scheduledconfigmgrd`
+6. `scheduledconfigmgrd` deletes all the configurations bound to the time range by publishing to the `APPL_DB`
+
+## High-Level Design
+
+- This feature will be a built-in part of SONiC, specifically inside the SWSS container.
```

Comment:
As discussed in the community meeting, can we explore designing this feature generically without disturbing the existing flows between app---appl_db---orchagent i.e just publish the configs at the time range required via new module(may be overlay DB instance) so that any of the existing internal design we have today wont be impacted.

For example,
Today, we do 
config-db---event-(vlan,intf,lag configs.)--mgr (vlanmgr/intf-mgmr)---appl_db---orchagent
config-db--event(e.g FRR configs)---bgpcfgd/frrrcfgd---FRR
config-db--event (e.g ACL)---orchagent

Without disturbing any of the existing design, can we explore options for handling the schedule manager out of the apps
For example,
config-db---scheduled-mgr---config event fires when the time range is active ---<above apps>

Reply:
i think simpler solution is to leave this to the admin see https://github.com/sonic-net/SONiC/pull/1700#discussion_r1639266665

Reply:
I was thinking about introducing a scheduler profile and map the profile to config(s) and when the config is stored into the config-DB, it checks for the profile association, if it's present, immediate publish is avoided and scheduler manager handles the configs that are associated with the profile and publishes the config event when the timer is expired based on the cron job. IMO, scheduler manager should work on top of apps and should not make any changes to the APPL_DB.

Reply:
so i think we agree that and conlicts and restore to default should be done by the admin setting up the time ranges and the configuration ( in your concept it the profile ), regarding which db to use, this is a complex issue, if we adopt your concept of using config_db than used cam change the configuration directly ( e.g. profile that set acl rule, now use can directly change the configuration... unless you introduce a more fundamental change that all configuration has a "profile id" which i think is a very big change, maybe application db is the not the correct DB, this relate to the question is scheduled configuration an application ? in any case i think config_db is not the right db to use.